ambry-rest/src/main/java/com.github.ambry.rest/NettyMessageProcessor.java

```diff
@@ -117,8 +120,10 @@ public void channelInactive(ChannelHandlerContext ctx) {
     logger.trace("Channel {} inactive", ctx.channel());
     nettyMetrics.channelDestructionRate.mark();
     if (request != null && request.isOpen()) {
-      logger.error("Request {} was aborted because the channel became inactive", request.getUri());
-      responseChannel.onResponseComplete(new ClosedChannelException());
+      logger.error("Request {} was aborted because the channel {} became inactive", request.getUri(), ctx.channel());
+      onRequestAborted(new ClosedChannelException());
+    } else {
+      close();
```

Comment:
The channel can be closed either by the remote end or the server. If the inactive was called because of the remote end, the `close()` call reconciles state on the server. I have added this detail to the documentation as suggested.